shared/js/moz_intl.js

```diff
@@ -84,7 +84,19 @@ global.mozIntl = {
 
     var intlFormat = new Intl.DateTimeFormat(locales, resolvedOptions);
 
+    // This is needed for a workaround for bug 1208808
+    // Remove when that bug is fixed
+    var hourFormatter;
+    if (options.dayperiod === true || options.dayperiod === false &&
+        intlFormat.resolvedOptions().hour12 === true) {
+      hourFormatter = Intl.DateTimeFormat(locales, {
+        hour: 'numeric',
+        hour12: false
+      });
+    }
+
     return {
+      resolvedOptions() { return resolvedOptions; },
       format: function(date, tokenFormats) {
```

Comment:
What's the accepted best practice:  should we allow for passing both `Date` objects and numbers representing milliseconds since epoch?